### PR TITLE
Provide ability to use all engines

### PR DESCRIPTION
**Pull request by Timoses**
* Ability to use all engines
* Documentation improvements
* Simplify Caster usage: It's no longer required to download the git repository itself.
    * See `getting_started` of the documentation
* Initial default configuration for Caster is now generated upon first start.

*NOTE: First #1  should be approved*

### CODE REVIEW

**Comment by Timoses** [comment]
Note that I was unable to test whether SAPI5 or Natlink work.

**Comment by LexiconCode** [comment]
I would recommend dropping 3.6. All other engines can utilize:
- Kaldi 3.7/3.8 - 64-bit
- Natlink Requires 3.7/3.8 - 32-bit 
- SAPI5 3.7/3.8 - 32-bit/64-bit 

 I've run into issues going lower than 3.7  which adds glue code for compatibility.

File: castervoice/core/controller.py [hunk 1]
```diff
@@ -1,13 +1,21 @@
 import logging
+import os
 import yaml
 
+try:
+    import importlib.resources as pkg_resources
+except ImportError:
+    # Try backported to PY<37 `importlib_resources`.
+    import importlib_resources as pkg_resources
```
Comment (LexiconCode) [inline castervoice/core/controller.py:9]: You could use [importlib-resources](https://pypi.org/project/importlib-resources/)

Comment (Timoses) [reply castervoice/core/controller.py:9]: I've added importlib-resources now..

Comment (Timoses) [reply castervoice/core/controller.py:9]: And will remove it again to drop support for v3.6

**Pull request merged**